File: 7-kyu/what-a--classy--song/python/solution.py
class Song:
    def __init__(self, title, artist):
        self.title = title
        self.artist = artist
        self.listeners = set()
    
    def how_many(self, todays_listeners):
        ret = 0
        for listener in todays_listeners:
            if listener.lower() not in self.listeners:
                ret += 1
                self.listeners.add(listener.lower())
        return ret

File: 7-kyu/what-a--classy--song/javascript/solution.js
// code here
class Song {
  constructor(title, artist) {
    this.title = title;
    this.artist = artist;
    this.listeners = [];
  }
  howMany(todaysListeners) {
    let ret = 0;
    for (const listener of todaysListeners) {
      if (!this.listeners.includes(listener.toLowerCase())) {
        ret++;
        this.listeners.push(listener.toLowerCase());
      }
    }
    return ret;
  }
}